pkg/monitor/monitor.go

```diff
@@ -228,9 +237,18 @@ func (m *Monitor) driverMonitor() {
 				// For any Running pod on that node using volume by the driver, kill the pod
 				// Degraded nodes are not considered offline and pods are not deleted from them.
 				if node.Status == volume.NodeOffline || node.Status == volume.NodeDegraded {
+					// Only initialize the map when it is absolutely necessary
+					if len(driverNodeToK8sNodeMap) == 0 {
+						driverNodeToK8sNodeMap = m.getVolumeDriverNodesToK8sNodeMap(nodes)
+					}
```

Comment:
Then in every cycle of 2mins we have to reconstruct it even if there are no offline nodes.
Also we can't take it outside of the outer for loop, since the data can become stale.

Took the approach of only initialising it if there is an offline node in every cycle.